### ISSUE TITLE
Poor results when using any quantizer

### ISSUE THREAD

**Issue by abhinavkulkarni** [facebookresearch/faiss#2198]
Hi,

TL;DR: The output from two runs is available here: 
Using flat index: https://jsitor.com/4O8-J-aFf
Using quantized index: https://jsitor.com/5LBMpVJwNa 

I am looking to build a compressed index (for reduced memory usage) from [BLINK](https://github.com/facebookresearch/BLINK) pre-computed encodings.

The BLINK model has two encoders (pre-trained). 
For a given text, the code first detects named-entity mentions and surrounding context. Those are passed through a context encoder to obtain detected mention encodings. Those are compared against (pre-computed) candidate encodings (these candidates are Wikipedia entities with the corresponding description). For each mention, top k nearest neighbors are obtained - this is where a FAISS index is useful as the number of Wiki candidates is in millions. 
A mention and corresponding top k candidates are jointly modeled by a crossencoder to produce final probability.

Candidate encodings are stored in `all_entities_large.t7` file (can be downloaded from [here](https://github.com/facebookresearch/BLINK/blob/main/download_blink_models.sh)).

I first build the dense indexes - `IndexFlat` and `IndexHNSW`.

I then passed a few of the candidates (and their descriptions) through the context encoder to obtain encodings. I was able to retrieve said candidates after searching dense indexes in top positions - in fact they were retrieved in the top most position.

Then I proceeded to build compressed indexes - `IVF256,PQ32x8`, `IVF512,PQ8x4`, etc. - I tried many combinations of `nlist`, `M` and `nbits` parameters. I even set the `nprobe` to the total number of Voronoi cells of coarse quantizer so that all cells are probed. However, I am unable to retrieve the candidates even in top 100 positions. I even built indexes where vectors are first transformed using PCA or OPQ, however no avail.

What am I missing?

I do understand that context and candidate encoders are slightly different models - so the vector they produce for the same mention/context pair is different - however I am able to retrieve the candidates using dense indexes, but not with quantized one.

What gives?

Thanks!
